src/services/ledger/index.types.d.ts

```diff
@@ -0,0 +1,37 @@
+type TxMetadata = {
+  type: LedgerTransactionType
+  pending: boolean
+  fee: Satoshis
+  feeUsd: number
+  sats: Satoshis
+  usd: number
+  currency: Currency
+}
+
+type AddLnTxReceiveMetadata = TxMetadata & {
+  hash: PaymentHash
+}
+
+type AddLnTxSendMetadata = TxMetadata & {
+  hash: PaymentHash
+  pubkey: Pubkey
+  feeKnownInAdvance: boolean
+}
+
+type AddLnIntraledgerTxSendMetadata = TxMetadata & {
+  hash: PaymentHash
+  pubkey: Pubkey
+  memoPayer: string | null
+  username: WalletName | null
+}
+
+type AddOnChainIntraledgerTxSendMetadata = TxMetadata & {
+  payee_addresses: OnChainAddress[]
+  sendAll: boolean
+  memoPayer: string | null
+  username: WalletName | null
```

Comment:
same question here

Reply:
Yes I think we should converge on a consistent representation of `absence of an expected value`. I don't really mind which we use as long as we are consistent. The discussion once came up with @samerbuna wether nor not to use `""` or `null` to represent `absence of a string value` and he said `null` was more idiomatic.

Either way I think `optional?` arguments should be avoided as I find this point towards there actually  being multiple  different use-cases rather than just the absence of a value.